lib/filename-generators/by-site-structure.js

```diff
@@ -7,16 +7,13 @@ module.exports = function generateFilename (resource, options) {
 	var extension = utils.getFilenameExtension(urlObject.pathname);
 	if (resource.isHtml() && !extension) {
 		var fileName = options.defaultFilename + (urlObject.search || '');
-		return getSafeAbsoluteFilename(path.join(urlObject.pathname || '/', fileName));
+		return getSafeRelativeFilename(path.join(urlObject.pathname || '/', fileName));
 	} else {
-		return getSafeAbsoluteFilename(urlObject.pathname);
+		return getSafeRelativeFilename(urlObject.pathname);
 	}
 };
 
-function getSafeAbsoluteFilename (urlPath){
+function getSafeRelativeFilename (urlPath){
 	urlPath = path.normalize(urlPath);
-
-	var safeFileName = urlPath.replace(/\.\./g, '.');
-	safeFileName = path.normalize(safeFileName);
-	return safeFileName;
+	return urlPath.substring(1).replace(/\.\./g, '.');
```

Comment:
We use urlPath which is the absolute path on the website, e.g.: `/some/path/something.html`. We want a relative path as a result, that is why we use `.substring(1)` to cut off the starting `/`.